### ISSUE TITLE
[java] ShortVariable false positive for java 22 unnamed variable `_`

### ISSUE THREAD

**Issue by emouty** [pmd/pmd#5892]
**Affects PMD Version:** 

**Rule:** [ShortVariable](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#shortvariable)

**Description:**
`_` is being flagged as shortvariable 

**Code Sample demonstrating the issue:**

```java
 switch (myValue) {
        case FirstClass _ -> throw new IllegalStateException("");
		case SecondClass _ -> System.out.print("I'm in second class");
		}
```

**Expected outcome:**

`_` should not be flagged as short variable since its a special variable name

**Running PMD through:** *Maven* | *PMD Intellij plugin*

**Comment by adangel** [comment]
Thanks for the report.
Btw - the feature is available since Java 22 and is called Unnamed Variable. See [JEP 456: Unnamed Variables & Patterns](https://openjdk.org/jeps/456) for details.
